simple-platform/src/main/java/org/simple/clinic/platform/crash/CrashReporter.kt

```diff
@@ -1,16 +1,40 @@
 package org.simple.clinic.platform.crash
 
 import android.app.Application
+import timber.log.Timber
 
-object CrashReporter: CrashReporter_Old {
+object CrashReporter : CrashReporter_Old {
+
+  private var sinks = emptyList<Sink>()
 
   override fun init(appContext: Application) {
+    // We will leave initialization to the individual sink constructor implementation
+    // since we do not know their requirements
+    // TODO: Remove this method in a later commit
   }
 
   override fun dropBreadcrumb(breadcrumb: Breadcrumb) {
+    sinks.forEach { sink ->
+      runSafely { sink.dropBreadcrumb(breadcrumb) }
+    }
   }
 
   override fun report(e: Throwable) {
+    sinks.forEach { sink ->
+      runSafely { sink.report(e) }
+    }
+  }
+
+  fun addSink(sink: Sink) {
```

Comment:
What are sinks and breadcrumbs here? Can you please point me to some article I can read to learn more about this? @vinaysshenoy

Reply:
A `Sink` is just a term for something which we can send data to (crash events, in this scenario).

A breadcrumb is additional information we can add to a crash, so that when a crash occurs, we can see this additional information as part of the crash report.

You can take a look at this (https://sentry.io/organizations/resolve-to-save-lives/issues/2358857128/?environment=production&project=1212614&query=is%3Aunresolved) crash report to see a list of breadcrumbs that are associated with it.